packages/core/admin/admin/src/core/apis/rbac.ts

```diff
@@ -0,0 +1,57 @@
+import { Location } from 'react-router-dom';
+
+import type { Permission, User } from '../../features/Auth';
+
+interface RBACContext extends Pick<Location, 'pathname' | 'search'> {
+  /**
+   * The current user.
+   */
+  user?: User;
+  /**
+   * The permissions of the current user.
+   */
+  permissions: Permission[];
+}
+
+interface RBACMiddleware {
+  (
+    ctx: RBACContext
+  ): (
+    next: (permissions: Permission[]) => Promise<Permission[]> | Permission[]
+  ) => (permissions: Permission[]) => Promise<Permission[]> | Permission[];
+}
+
+class RBAC {
+  private middlewares: RBACMiddleware[] = [];
+
+  constructor() {}
+
+  use(middleware: RBACMiddleware[]): void;
+  use(middleware: RBACMiddleware): void;
```

Comment:
```suggestion
```
Do we need those? We've got the same types below

Reply:
OVERLOADS